shared/src/k8s/mod.rs

```diff
@@ -621,6 +622,58 @@ impl KubeInterface for KubeImpl {
     }
 }
 
+/// This deletes an Instance unless it has already been deleted by another node
+/// or fails after multiple retries.
+pub async fn try_delete_instance(
+    kube_interface: &dyn KubeInterface,
+    instance_name: &str,
+    instance_namespace: &str,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    for x in 0..MAX_INSTANCE_UPDATE_TRIES {
```

Comment:
Do we really need to retry even for a delete operation?

Reply:
I was thinking we needed to in case there is an error reaching api server. Do you expect this to not be the case for deletes?